### ISSUE TITLE
[HOLD for payment 2024-10-24] De-dupe `OpenApp` requests.

### ISSUE THREAD

**Issue by roryabraham** [Expensify/App#50076]
### Problem
We are seeing some scenarios where multiple `OpenApp` requests are serialized in a row. When multiple such requests are sent, it's redundant, slows down the app, and contributes to server traffic.

### Solution
De-dupe `OpenApp` requests such that only one can be serialized at a time.

<details><summary>Issue Owner</summary>Current Issue Owner: @trjExpensify</details>

**Comment by roryabraham** [comment]
cc @gedu

**Comment by gedu** [comment]
Hey, I'm Edu from Callstack, I can take this one

**Comment by roryabraham** [comment]
Going on parental leave. @gedu @mountiny @shubham1206agra I leave this in your capable hands. My parting note is to please use as many automated tests to cover this change as possible 👍🏼

**Comment by mountiny** [comment]
Thanks @gedu

**Comment by mountiny** [comment]
$250 to @shubham1206agra

**Comment by trjExpensify** [comment]
Payment due on the 24th Melv, chill mate.

**Comment by trjExpensify** [comment]
👋 @shubham1206agra can we get a checklist here please? Thanks! 

**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@shubham1206agra] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@shubham1206agra] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@shubham1206agra] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@shubham1206agra] Determine if we should create a regression test for this bug.
- [ ] [@shubham1206agra] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@trjExpensify] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by trjExpensify** [comment]
Hm, it doesn't sound like a new feature. @mountiny can you confirm that?

**Comment by trjExpensify** [comment]
^^ waiting for a response on that to confirm.

**Comment by mountiny** [comment]
I think this one is on the edge between a new feature and a bug. Essentially, it just worked as we designed the Sequential Queue, so this behaviour was expected. Later, we saw that this can cause unwanted performance bottlenecks, so we decided to change this behaviour. 

So I agree its not necessary to be identifying the original PRs etc because the authors who worked on this know about this change.

@shubham1206agra your test is not complete, where is the tester checking for expected behaviour?
